http/handler.go

```diff
@@ -598,6 +599,43 @@ func parseRequest(perfStandby bool, r *http.Request, w http.ResponseWriter, out
 	return nil, err
 }
 
+// parseFormRequest parses values from a form POST.
+//
+// A nil map will be returned if the format is empty or invalid.
+func parseFormRequest(r *http.Request) (map[string]interface{}, error) {
+	maxRequestSize := r.Context().Value("max_request_size")
+	if maxRequestSize != nil {
+		max, ok := maxRequestSize.(int64)
+		if !ok {
+			return nil, errors.New("could not parse max_request_size from request context")
+		}
+		if max > 0 {
+			r.Body = ioutil.NopCloser(io.LimitReader(r.Body, max))
+		}
+	}
+	if err := r.ParseForm(); err != nil {
+		return nil, err
+	}
+
+	var data map[string]interface{}
+
+	if len(r.PostForm) != 0 {
+		data = make(map[string]interface{}, len(r.PostForm))
+		for k, v := range r.PostForm {
+			switch len(v) {
+			case 0, 1:
+				data[k] = v[0]
```

Comment:
@ncabatoff Indeed! I'd noticed this when I was first bringing in some similar code and forgot to double check it. The curious part was why the tests were passing... and it turns out that the query processing ends up just stripping the zero length values entirely so that case wasn't getting hit. I've now added explicit handling.